### ISSUE TITLE
Not possible to download on server which break connection

### ISSUE THREAD

**Issue by trivalik** [bezzad/Downloader#177]
I tried to download from server which break connection all the time. I know on the early 2000 we used programs that was just restarting then. i.e. URL for test purpose https://zorin.mirrors.itworxx.de/17/Zorin-OS-17.2-Education-64-bit.iso

**Comment by bezzad** [comment]
This issue originated from the server-side. I attempted to download it using the Chrome browser, but it resulted in an exception: Network Error.
![image](https://github.com/user-attachments/assets/019c1e7c-3784-4f8f-b765-d397161d8fc7)

**Comment by trivalik** [comment]
Yes you are right. But in case this downloader supports retry, then it should retry. Up to you.

**Comment by bezzad** [comment]
@trivalik The Downloader retries when an exception occurs. However, if the number of retries exceeds the MaxTryAgainOnFailover value, it throws an exception and stops the download. I checked the download process for this URL, and the retry count was very high. When I set DownloadConfiguration.MaxTryAgainOnFailover = 10000, I was able to download the file successfully. By default, this value is set to int.MaxValue.

Additionally, I should mention that the [Downloader.Desktop](https://github.com/bezzad/Downloader.Desktop) repository is for a project built with Avalonia as a UI for the Downloader package. It is still under development and has a long way to go before completion.

**Comment by trivalik** [comment]
Sorry for disturbing, my fault. I wonder that you got no `StackOverflowException`, since on every retry the stackstrace grows by 2 entries.
